src/Illuminate/Console/GeneratorCommand.php

```diff
@@ -417,13 +417,7 @@ protected function sortImports($stub)
      */
     protected function getNameInput()
     {
-        $name = trim($this->argument('name'));
-
-        if (Str::endsWith($name, '.php')) {
-            return Str::substr($name, 0, -4);
-        }
-
-        return $name;
+        return (string) Str::of($this->argument('name'))->trim()->beforeLast('.php');
```

Comment:
You could've called `toString()` and that would've passed the result to a string value.

Reply:
@gianfranco-rocco it's `__toString()`  not `toString()`, so would've looked less clean.

Reply:
There's actually a `toString` method on the `Stringable` class. Here's the link to the [API doc](https://laravel.com/api/11.x/Illuminate/Support/Stringable.html#method_toString).